aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -171,7 +193,18 @@ Tensor empty_like(const Tensor& self, const TensorOptions& options) {
                                        self.q_scale().toDouble(),
                                        self.q_zero_point().toLong());
   }
-  return at::empty(self.sizes(), options);
+
+  auto memory_format = optional_memory_format.value_or(MemoryFormat::Contiguous);
+  auto use_memory_format = memory_format;
+  if (memory_format == MemoryFormat::Preserve) {
+    if (self.is_contiguous(MemoryFormat::ChannelsLast)) {
```

Comment:
This is not so bad when we only have two memory formats,  but what's going to happen when we add more?

Reply:
Added error message